src/main/java/net/dv8tion/jda/api/sharding/DefaultShardManager.java

```diff
@@ -509,16 +509,16 @@ protected JDAImpl buildInstance(final int shardId) throws LoginException, Interr
         {
             try
             {
-                Pair<String, Integer> gateway = jda.getGatewayBot();
-                this.gatewayURL = gateway.getLeft();
+                SessionController.GatewayBot gateway = jda.getGatewayBot();
```

Comment:
Does this actually compile? It looks like `getGatewayBot` is still returning a `Pair`

Reply:
You're confusing `SessionController#getGatewayBot` and `JDAImpl#getGatewayBot`